libparsec/crates/types/tests/unit/manifest.rs

```diff
@@ -39,3 +44,259 @@ fn invalid_deserialize_data(#[case] data: &[u8], #[case] error: DataError) {
+        "9013014ce66af9ea27acdabd278345b1a13ee69107c2419b2991268048a0bf51e472fa6f02"
+        "0767a0c463d3d6a61f2c855fdcbc0073734fdf"
+    );
+    let signed_encrypted = hex!(
+        "be4be31375b99ebf3e76c54b1e91d05f5739e7e11b631b0c51cf2a91c3ca92c3a074e4d044"
+        "9dfd02c55444d9bc500402680daa04bcbf13d8d38f8a296597cb31cc2bc277c7d1c8bb5bb4"
+        "2321b134055351d9a81ce423d42c85ede0cf8bc8722662ce1227e4bef0645c7ae934df708b"
+        "397cdc582d6eabe8155cf3a00bde4cc2f2ff58e9bd415829fb4e985bc5c170b94bfe05378c"
+        "03bd86a5c074eacef07dfb5760fe6b072960c70d7ca95c9a1ec6ab5769e4bd700d7f31258a"
+        "e24e5a8f1525950fea50d447082c90e4599d30833d2776b268b0b59804ead23967e5ef3e2c"
+        "f86ad132cc806f23bfcc2be8a2daae49139f9a3fb0a6cf53a3affa839e2b3ee8fae63fc991"
+        "d4b644954559695f0c61219e0551edec7123466a20e9"
+    );
+
+    let now = "2021-12-04T11:50:43.208821Z".parse().unwrap();
+    let parent = EntryID::from_hex("09e679aa6e1147fbaa5580073202fc7f").unwrap();
+    let id = EntryID::from_hex("87c6b5fd3b454c94bab51d6af1c6930b").unwrap();
+
+    let expected_file_manifest = FileManifest {
+        author: alice.device_id.clone(),
+        timestamp: now,
+        id,
+        parent,
+        version: 0,
+        created: now,
+        updated: now,
+        size: 0,
+        blocksize: Blocksize::try_from(8).unwrap(),
+        blocks: vec![],
+    };
+
+    assert!(expected_file_manifest
+        .verify(&alice.device_id, now, Some(id), Some(0))
+        .is_ok());
+
+    assert_eq!(
+        FileManifest::verify_and_load(
+            &signed,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0),
+        )
+        .unwrap(),
+        expected_file_manifest
+    );
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &signed_encrypted,
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0),
+        )
+        .unwrap(),
+        expected_file_manifest
+    );
+
+    // Also test round trip
+    assert_eq!(
+        FileManifest::verify_and_load(
+            &expected_file_manifest.dump_and_sign(&alice.signing_key),
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0),
+        )
+        .unwrap(),
+        expected_file_manifest
+    );
+    assert_eq!(
+        FileManifest::decrypt_verify_and_load(
+            &expected_file_manifest.dump_sign_and_encrypt(&alice.signing_key, &alice.local_symkey),
+            &alice.local_symkey,
+            &alice.verify_key(),
+            &alice.device_id,
+            now,
+            Some(id),
+            Some(0),
+        )
+        .unwrap(),
+        expected_file_manifest
+    );
+}
+
+#[rstest]
+fn invalid_load(alice: &Device) {
+    // The blob is generated from the test `dump_load`
```

Comment:
`dump_load` may change in the future, in that case the comment will silently become irrelevant. Hence why it's better to add a comment about the actual content